caas/kubernetes/provider/k8s_test.go

```diff
@@ -278,6 +281,51 @@ func (s *K8sBrokerSuite) TestConfig(c *gc.C) {
 	c.Assert(s.broker.Config(), jc.DeepEquals, s.cfg)
 }
 
+func (s *K8sBrokerSuite) TestSetConfig(c *gc.C) {
+	ctrl := s.setupBroker(c)
+	defer ctrl.Finish()
+
+	err := s.broker.SetConfig(s.cfg)
+	c.Assert(err, jc.ErrorIsNil)
+}
+
+func (s *K8sBrokerSuite) TestControllerNamespaceRenaming(c *gc.C) {
+	cfg, err := config.New(config.UseDefaults, testing.FakeConfig().Merge(testing.Attrs{
+		config.NameKey: "controller",
+	}))
+	c.Assert(err, jc.ErrorIsNil)
+	s.cfg = cfg
+
+	ctrl := s.setupBroker(c)
+	defer func() {
+		ctrl.Finish()
+		// reset s.cfg
+		s.SetUpSuite(c)
+	}()
+
+	c.Assert(s.broker.GetCurrentNamespace(), jc.DeepEquals, "controller"+"-"+s.cfg.UUID())
+
+}
+
+func (s *K8sBrokerSuite) TestBootstrap(c *gc.C) {
+	ctrl := s.setupBroker(c)
+	defer ctrl.Finish()
+
+	ctx := envtesting.BootstrapContext(c)
+	result, err := s.broker.Bootstrap(ctx,
+		&context.CloudCallContext{},
+		environs.BootstrapParams{
+			ControllerConfig:     testing.FakeControllerConfig(),
+			BootstrapSeries:      "bionic",
```

Comment:
BootstrapSeries is not relevant for caas, and especially not "bionic"
We should error if they set it